### PR TITLE
ADEN-4682 Log VAST details from selected Vulcan netwroks and advertisers

### PR DESCRIPTION
**Pull request by fraszczakszymon**

### CODE REVIEW

**Comment by WikiaTech** [comment]
SonarQube analysis reported 7 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [AdEngine2Hooks.class.php#L24](https://github.com/Wikia/app/blob/96f04a2f77e07dc4e69cf98c38c6b921305a5d68/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L24): Remove the unused function parameters "$title, $article, $output, $user, $wiki". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [AdEngine2Hooks.class.php#L99](https://github.com/Wikia/app/blob/96f04a2f77e07dc4e69cf98c38c6b921305a5d68/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L99): Remove the unused function parameter "$scripts". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [AdEngine2Hooks.class.php#L204](https://github.com/Wikia/app/blob/96f04a2f77e07dc4e69cf98c38c6b921305a5d68/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L204): Remove the unused function parameter "$skin". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [AdEngine2Hooks.class.php#L229](https://github.com/Wikia/app/blob/96f04a2f77e07dc4e69cf98c38c6b921305a5d68/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L229): Remove the unused function parameters "$jsExtensionPackages, $scssPackages". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [playerTracker.js#L12](https://github.com/Wikia/app/blob/96f04a2f77e07dc4e69cf98c38c6b921305a5d68/extensions/wikia/AdEngine/js/video/player/playerTracker.js#L12): Function has 9 parameters which is greater than 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. MAJOR [playerTracker.js#L26](https://github.com/Wikia/app/blob/96f04a2f77e07dc4e69cf98c38c6b921305a5d68/extensions/wikia/AdEngine/js/video/player/playerTracker.js#L26): Function has a complexity of 12 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)
1. INFO [AdEngine2Hooks.class.php#L27](https://github.com/Wikia/app/blob/96f04a2f77e07dc4e69cf98c38c6b921305a5d68/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L27): Complete the task associated to this "TODO" comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)

File: extensions/wikia/AdEngine/js/video/player/porvata/vastLogger.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+	}
+
+	function logVast(player, params, data) {
+		var configKey,
+			eventName = data['event_name'] || '',
+			vulcanAdvertiser = data['vulcan_advertiser'] || '',
+			vulcanNetwork = data['vulcan_network'] || '',
+			postData;
+
+		configKey = [ vulcanNetwork, vulcanAdvertiser, eventName ].join('_');
```
Comment (v3rth) [inline extensions/wikia/AdEngine/js/video/player/porvata/vastLogger.js]: Maybe move this stuff above to some function (``createConfigKey``?). I know that you are using ``eventName`` and other variables below, but this function looks too big for me

File: extensions/wikia/AdEngine/js/video/player/porvata/vastLogger.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+		if (config.indexOf(configKey) === -1) {
+			return;
+		}
+
+		postData = [
```
Comment (v3rth) [inline extensions/wikia/AdEngine/js/video/player/porvata/vastLogger.js]: Same for this (extract), but I'm not sure about it (it might be overkill :smile:)

**Review by v3rth — APPROVED** [review]
LGTM

**Comment by WikiaTech** [comment]
SonarQube analysis reported 7 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [AdEngine2Hooks.class.php#L24](https://github.com/Wikia/app/blob/ca9b85a2933aa14636a3a2da4e6bc0926bf6023b/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L24): Remove the unused function parameters "$title, $article, $output, $user, $wiki". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [AdEngine2Hooks.class.php#L99](https://github.com/Wikia/app/blob/ca9b85a2933aa14636a3a2da4e6bc0926bf6023b/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L99): Remove the unused function parameter "$scripts". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [AdEngine2Hooks.class.php#L204](https://github.com/Wikia/app/blob/ca9b85a2933aa14636a3a2da4e6bc0926bf6023b/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L204): Remove the unused function parameter "$skin". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [AdEngine2Hooks.class.php#L229](https://github.com/Wikia/app/blob/ca9b85a2933aa14636a3a2da4e6bc0926bf6023b/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L229): Remove the unused function parameters "$jsExtensionPackages, $scssPackages". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. MAJOR [playerTracker.js#L12](https://github.com/Wikia/app/blob/ca9b85a2933aa14636a3a2da4e6bc0926bf6023b/extensions/wikia/AdEngine/js/video/player/playerTracker.js#L12): Function has 9 parameters which is greater than 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. MAJOR [playerTracker.js#L26](https://github.com/Wikia/app/blob/ca9b85a2933aa14636a3a2da4e6bc0926bf6023b/extensions/wikia/AdEngine/js/video/player/playerTracker.js#L26): Function has a complexity of 12 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)
1. INFO [AdEngine2Hooks.class.php#L27](https://github.com/Wikia/app/blob/ca9b85a2933aa14636a3a2da4e6bc0926bf6023b/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L27): Complete the task associated to this "TODO" comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)

**Pull request merged**